src/bindings/config.ts

```diff
@@ -79,28 +79,127 @@ export class Config implements IConfig {
         }
       }
 
+      this.safeSet(
+        () => {
+          if (configRef) this.addon.configSetHttpsOnly(configRef, options.httpsOnly as boolean);
```

Comment:
at the beginning of the initialize function, check if we get configRef or not. if not, throw error or return error whatever is suitable